src/components/AddPost/index.js

```diff
@@ -1,35 +1,30 @@
 import { Button } from "@ahaui/react";
 import React from "react";
-import { useForm } from "react-hook-form";
 import { useDispatch } from "react-redux";
-import Layout from "../components/common/Layout";
-import PostForm from "../components/Manage/PostForm";
+import { Link, useNavigate } from "react-router-dom";
 import styled from "styled-components";
-import { Link } from "react-router-dom";
+import { createPostsAction } from "../../store/reducers/postsReducer";
+import Layout from "../common/Layout";
+import PostForm from "../common/PostForm";
 
-export default function AddPost() {
-  let loading;
+export default function AddPost(props) {
+  let history = useNavigate();
   const dispatch = useDispatch();
-  const {
-    register,
-    handleSubmit,
-    watch,
-    formState: { errors },
-  } = useForm();
 
-  const onSubmitAddPost = ({ title, content, image }) => {
+  const onSubmitAddPost = async ({ title, content, picture }) => {
     console.log({
       title,
       content,
-      image,
+      picture,
     });
-    // TODO: Add dispatch action
-    // dispatch(
-    //   loginAction({
-    //     title,
-    //     content,
-    //   })
-    // );
+    await dispatch(
+      createPostsAction({
+        title,
+        content,
+        picture,
+      })
+    ).unwrap();
+    history("/manage");
```

Comment:
If `createPostsAction` fails, I don't think it's a great idea to push to `/manage`